modules/ip/datapath/arp_output_request.c

```diff
@@ -115,34 +113,10 @@ arp_output_request_process(struct rte_graph *graph, struct rte_node *node, void
 }
 
 static void arp_output_request_register(void) {
-	nexthop_ring = rte_ring_create(
-		"arp_output_request",
-		RTE_GRAPH_BURST_SIZE * 4,
-		SOCKET_ID_ANY,
-		RING_F_MP_RTS_ENQ | RING_F_MC_RTS_DEQ
-	);
-	if (nexthop_ring == NULL)
-		ABORT("rte_ring_create(arp_output_request): %s", rte_strerror(rte_errno));
-
-	arp_pool = rte_pktmbuf_pool_create(
-		"arp_output_request",
-		RTE_GRAPH_BURST_SIZE * 4,
-		256, // cache_size
-		0, // priv_size
-		RTE_MBUF_DEFAULT_BUF_SIZE,
-		SOCKET_ID_ANY
-	);
-	if (arp_pool == NULL)
-		ABORT("rte_pktmbuf_pool_create(arp_output_request): %s", rte_strerror(rte_errno));
-}
-
-static void arp_output_request_unregister(void) {
-	rte_ring_free(nexthop_ring);
-	rte_mempool_free(arp_pool);
+	gr_slow_path_input_add_handler(RTE_ETHER_TYPE_IPV4, "arp_output_request");
```

Comment:
@rjarry Not sure if I can use directly `RTE_ETHER_TYPE_IPV4`, add an enum in `gr_slow_path.h`, or create an API to fetch the next available slot ?
`arp_slow_path_type = gr_slow_path_input_add_handler ("arp_output_request");`
And then `slow_path_post(arp_slow_path_type, ... )`

Reply:
You will probably need to do it in two steps. First append the node to slow_path next nodes, and then, add the node id indexed with the slow path "type", in this case `ARP_SOLICIT`.

Have a look at how dynamic nodes are registered after ip_local or after eth_input.

Also, I think we need a local enum here to reference the slow path type. Not reuse `RTE_ETHER_TYPE_IPV4` which is an ether type value for IPv4, not for ARP (maybe a typo?).

Reply:
This is done in `gr_slow_path_input_add_handler`, so we don't have to append explicitly the next node.
As we abort anyway if this fails, I didn't see the value of asking the user to attach manually the next node.

Maybe we can also simplify the API for `gr_eth_input_add_type`, so it takes a node name instead of an edge ?
`void gr_eth_input_add_type(rte_be16_t eth_type, char * node_name)`